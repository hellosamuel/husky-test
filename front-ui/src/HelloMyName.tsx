import { memo, useMemo } from 'react'

export function getUppercaseName(name: string) {
  const first = name[0].toUpperCase()
  let result = first
  for (let i = 1; i < name.length; i += 1) {
    result += name[i]
  }
  return result
}

type HelloMyNameProps = {
  name?: string
}

function HelloMyName({ name = 'nobody' }: HelloMyNameProps) {
  const userName = useMemo(() => getUppercaseName(name), [name])
  return (
    <p>
      Hello! <strong>{userName}</strong>
    </p>
  )
}

export default memo(HelloMyName)
