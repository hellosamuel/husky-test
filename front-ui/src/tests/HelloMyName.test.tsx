import { render, screen } from '@testing-library/react'
import HelloMyName, { getUppercaseName } from '../HelloMyName'

test('Render HelloMyName correctly', () => {
  const { asFragment } = render(<HelloMyName />)

  expect(asFragment()).toMatchSnapshot()
})

test('Should be name correctly (uppercase)', () => {
  const name = 'user name'
  render(<HelloMyName name={name} />)

  const expectedName = 'User name'
  expect(getUppercaseName(name)).toBe(expectedName)

  const nameElm = screen.queryByText(expectedName)
  expect(nameElm).toBeInTheDocument()
})

test('Should be name correctly (default name)', () => {
  render(<HelloMyName />)

  const expectedName = 'Nobody'
  const nameElm = screen.queryByText(expectedName)
  expect(nameElm).toBeInTheDocument()
})
