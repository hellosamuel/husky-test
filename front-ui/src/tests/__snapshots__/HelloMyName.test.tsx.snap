// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render HelloMyName correctly 1`] = `
<DocumentFragment>
  <p>
    Hello! 
    <strong>
      Nobody
    </strong>
    . Welcome to Husky Test.
  </p>
</DocumentFragment>
`;
